server/asset/dcr/dcr.go

```diff
@@ -234,10 +220,10 @@ func (dcr *DCRBackend) transaction(txHash *chainhash.Hash) (*Tx, error) {
 	var sumIn, sumOut uint64
 	// Parse inputs and outputs, grabbing only what's needed.
 	inputs := make([]txIn, 0, len(verboseTx.Vin))
-	for _, input := range verboseTx.Vin {
-		if input.Txid == "" {
-			inputs = append(inputs, txIn{vout: input.Vout})
-			continue
+	isCoinbase := false
+	for vin, input := range verboseTx.Vin {
+		if vin == 0 && input.Coinbase != "" || input.Stakebase != "" {
```

Comment:
I'm not totally sure we need the `vin == 0` check if `.Coinbase` and `.Stakebase` tell us.
But did you want the terms grouped like `vin == 0 && (input.Coinbase != "" || input.Stakebase != "")`?